src/analysis/retail/rogue/outlaw/modules/talents/BladeRush.tsx

```diff
@@ -15,44 +33,143 @@ class BladeRush extends Analyzer {
   protected abilities!: Abilities;
   protected spellUsable!: SpellUsable;
 
+  unusableUptimes: Uptime[] = [];
+  isInUsablePeriod: boolean = false;
+  castEntries: BoxRowEntry[] = [];
+
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS.BLADE_RUSH_TALENT);
+
+    this.addEventListener(Events.resourcechange.by(SELECTED_PLAYER), this.onResourceChange);
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.BLADE_RUSH_TALENT),
+      this.onBladeRush,
+    );
+
     this.addEventListener(
-      Events.cast
-        .by(SELECTED_PLAYER)
-        .spell([
-          SPELLS.DISPATCH,
-          SPELLS.EVISCERATE,
-          SPELLS.KIDNEY_SHOT,
-          SPELLS.BETWEEN_THE_EYES,
-          SPELLS.SLICE_AND_DICE,
-        ]),
-      this.onFinishMove,
+      Events.UpdateSpellUsable.by(SELECTED_PLAYER).spell(TALENTS.BLADE_RUSH_TALENT),
+      this.onBladeRushUsable,
     );
+    
+    this.unusableUptimes.push({
+      start: this.owner.fight.start_time,
+      end: -1,
+    });
+  }
+
+  endUsablePeriod(timestamp: number) {
```

Comment:
It's not really an issue, but consider marking functions not intended to be consumed from outside of this module as `private`.